xcube/core/resampling/rectify.py

```diff
@@ -634,37 +642,52 @@ def _compute_var_image_numpy(
     dst_height = dst_src_ij_images.shape[-2]
     dst_shape = src_var.shape[:-2] + (dst_height, dst_width)
     dst_values = np.full(dst_shape, fill_value, dtype=src_var.dtype)
+    src_bbox = [0, 0, src_var.shape[-2], src_var.shape[-1]]
     _compute_var_image_numpy_parallel(
-        src_var, dst_src_ij_images, dst_values, interpolation
+        src_var, dst_src_ij_images, dst_values, src_bbox, interpolation
     )
     return dst_values
 
 
-@nb.njit(nogil=True, cache=True)
 def _compute_var_image_xarray_dask_block(
-    src_var_image: np.ndarray,
     dst_src_ij_images: np.ndarray,
+    src_var_image: xr.DataArray,
     fill_value: Union[int, float, complex],
     interpolation: int,
+    chunksize: tuple[int],
 ) -> np.ndarray:
     """Extract source pixels from np.ndarray source
     and return a block of a dask array.
     """
     dst_width = dst_src_ij_images.shape[-1]
     dst_height = dst_src_ij_images.shape[-2]
     dst_shape = src_var_image.shape[:-2] + (dst_height, dst_width)
+    dst_out = np.full(chunksize, fill_value, dtype=src_var_image.dtype)
+    if np.all(np.isnan(dst_src_ij_images[0])):
+        return dst_out
     dst_values = np.full(dst_shape, fill_value, dtype=src_var_image.dtype)
+    src_bbox = [
+        int(np.nanmin(dst_src_ij_images[0])),
+        int(np.nanmin(dst_src_ij_images[1])),
+        min(int(np.nanmax(dst_src_ij_images[0])) + 2, src_var_image.shape[-1]),
+        min(int(np.nanmax(dst_src_ij_images[1])) + 2, src_var_image.shape[-2]),
+    ]
+    src_var_image = src_var_image[
+        ..., src_bbox[1] : src_bbox[3], src_bbox[0] : src_bbox[2]
+    ].values.astype(np.float64)
     _compute_var_image_numpy_sequential(
-        src_var_image, dst_src_ij_images, dst_values, interpolation
+        src_var_image, dst_src_ij_images, dst_values, src_bbox, interpolation
     )
-    return dst_values
+    dst_out[..., :dst_height, :dst_width] = dst_values
+    return dst_out
 
 
 @nb.njit(nogil=True, parallel=True, cache=True)
 def _compute_var_image_numpy_parallel(
     src_var_image: np.ndarray,
     dst_src_ij_images: np.ndarray,
     dst_var_image: np.ndarray,
+    src_bbox: list[int],
```

Comment:
```suggestion
    src_bbox: tuple[int, int, int, int],
```

or

```suggestion
    src_bbox: np.ndarray,
```

See comment above.